### PR TITLE
[196] Implementation of InvocationExpression in export to textual files

### PR DESCRIPTION
**Pull request by wbilem**

### CODE REVIEW

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 1]
```diff
@@ -385,8 +386,20 @@ public String caseOperatorExpression(OperatorExpression op) {
...
     public String caseInvocationExpression(InvocationExpression expression) {
-        LOGGER.warn("InvocationExpression are not handled yet");
-        return "";
+        Appender builder = this.newAppender();
+        List<Relationship> relationships = expression.getOwnedRelationship();
+        if (!relationships.isEmpty() && (relationships.get(0) instanceof FeatureTyping || relationships.get(0) instanceof Subsetting)) {
+            relationships.stream()
+                .filter(Specialization.class::isInstance)
+                .map(Specialization.class::cast)
+                .filter(specialization -> specialization instanceof FeatureTyping || specialization instanceof Subsetting)
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:395]: You already check in the _if_ statement that the `relationships ` contains a _FeatureTyping_ or a _Subsetting_ in first position.
So I think this `.filter(...)` could be removed.

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 2]
```diff
@@ -775,6 +788,47 @@ private void appendArgumentMember(Appender builder, ParameterMembership paramete
...
+        builder.append(LabelConstants.OPEN_PARENTHESIS);
+        Relationship relationship = expression.getOwnedRelationship().stream()
+                   .filter(FeatureMembership.class::isInstance)
+                   .map(FeatureMembership.class::cast)
+                   .findFirst()
+                   .orElse(null);
+        if (relationship instanceof ParameterMembership) {
+            appendPositionalArgumentList(builder, expression);
+        }
+        else if (relationship instanceof FeatureMembership) {
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:802]: `else if` => `} else if`

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 3]
```diff
@@ -775,6 +788,47 @@ private void appendArgumentMember(Appender builder, ParameterMembership paramete
...
+            .filter(ParameterMembership.class::isInstance)
+            .map(ParameterMembership.class::cast)
+            .filter(param -> {
+                if (param.getOwnedMemberParameter() != null) {
+                    return param.getOwnedMemberParameter().getOwnedRelationship().stream()
+                    .filter(FeatureValue.class::isInstance)
+                    .map(FeatureValue.class::cast)
+                    .findAny().isPresent();
+                }
+                else {
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:819]: `else` => `} else`

Comment (AxelRICHARD) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:819]: Please check everywhere else in this PR for similar cases

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 4]
```diff
@@ -775,6 +788,47 @@ private void appendArgumentMember(Appender builder, ParameterMembership paramete
...
+    }
+    
+    private void appendPositionalArgumentList(Appender builder, InvocationExpression expression) {
+        List<ParameterMembership> parameterMemberships = expression.getOwnedRelationship().stream()
+            .filter(ParameterMembership.class::isInstance)
+            .map(ParameterMembership.class::cast)
+            .filter(param -> {
+                if (param.getOwnedMemberParameter() != null) {
+                    return param.getOwnedMemberParameter().getOwnedRelationship().stream()
+                    .filter(FeatureValue.class::isInstance)
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:815]: format problem here

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 5]
```diff
@@ -775,6 +788,47 @@ private void appendArgumentMember(Appender builder, ParameterMembership paramete
...
+                    .map(FeatureValue.class::cast)
+                    .findAny().isPresent();
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:817]: `isPresent()` should be on the next line

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java (hunk 1 above)
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java (hunk 2 above)
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java (hunk 3 above)
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java (hunk 4 above)
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java]: done

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java (hunk 5 above)
Comment (wbilem) [reply backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java]: done

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**